homeassistant/components/trafikverket_camera/__init__.py

```diff
@@ -22,6 +29,9 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Trafikverket Camera from a config entry."""
 
+    if entry.unique_id and entry.unique_id.startswith(DOMAIN):
```

Comment:
I think it's actually fine to keep the hard config entry migration in this case. I was a bit fast in the earlier review. Sorry.